libraries/chain/block_header_state.cpp

```diff
@@ -24,8 +24,12 @@ digest_type block_header_state::compute_base_digest() const {
    assert(active_proposer_policy);
    fc::raw::pack( enc, *active_proposer_policy );
 
-   fc::raw::pack( enc, latest_proposed_proposer_policy );
-   fc::raw::pack( enc, latest_pending_proposer_policy );
+   if (latest_proposed_proposer_policy) {
+      fc::raw::pack( enc, *latest_proposed_proposer_policy );
+   }
+   if (latest_pending_proposer_policy) {
+      fc::raw::pack( enc, *latest_pending_proposer_policy );
+   }
```

Comment:
why this change? I believe we should pack the `shared_ptr` itself.

Reply:
We don't pack `activated_protocol_features` (a shared ptr) if it does not exist. We also pack the actual content of `active_proposer_policy` (we assert it is always present).

Reply:
For a digest it doesn't matter, but if we were packing for writing we would want the shared_ptr written so that we could unpack it. Otherwise, you would not know that entry from the next. 

Seems we should be consistent with `activated_protocol_features`. Curious if @arhag has thoughts on this.

Reply:
I talked to @arhag about this. 

- Even though this is for a digest, it is good practice to keep it consistent with what you would do for a packed that you would write. 
- `activated_protocol_features` should always be present. We can change that to an `assert` and just write it out.
- For things that are optionally present we should always pack the bool indicating if they are there. So that means packing the shared_ptr, not the thing itself.